neptune/instance.go

```diff
@@ -2,38 +2,160 @@ package neptune
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	"github.com/ONSdigital/dp-dimension-importer/model"
+	"github.com/ONSdigital/dp-graph/neptune/query"
+	"github.com/ONSdigital/go-ns/log"
+	gremgo "github.com/ONSdigital/gremgo-neptune"
+	"github.com/pkg/errors"
 )
 
+const codeListNotFoundFmt = "VertexStep(OUT,[usedBy],vertex), HasStep([~label.eq(_code_list_%s)"
+
+// CountInsertedObservations returns the current number of observations relating to a given instance
 func (n *NeptuneDB) CountInsertedObservations(ctx context.Context, instanceID string) (count int64, err error) {
-	return 0, nil
+	return n.getNumber(fmt.Sprintf(query.CountObservations, instanceID))
 }
 
+// AddVersionDetailsToInstance updates an instance node to contain details of which
+// dataset, edition and version the instance will also be known by
 func (n *NeptuneDB) AddVersionDetailsToInstance(ctx context.Context, instanceID string, datasetID string, edition string, version int) error {
+	data := log.Data{
+		"instance_id": instanceID,
+		"dataset_id":  datasetID,
+		"edition":     edition,
+		"version":     version,
+	}
+
+	q := fmt.Sprintf(query.AddVersionDetailsToInstance, instanceID, datasetID, edition, version)
+
+	if _, err := n.exec(q); err != nil {
+		log.ErrorC("neptune exec failed on AddVersionDetailsToInstance", err, data)
+		return err
+	}
 	return nil
 }
 
+// SetInstanceIsPublished sets a flag on an instance node to indicate the published state
 func (n *NeptuneDB) SetInstanceIsPublished(ctx context.Context, instanceID string) error {
+	data := log.Data{
+		"instance_id": instanceID,
+	}
+
+	q := fmt.Sprintf(query.SetInstanceIsPublished, instanceID)
+
+	if _, err := n.exec(q); err != nil {
+		log.ErrorC("neptune exec failed on SetInstanceIsPublished", err, data)
+		return err
+	}
 	return nil
 }
 
+// CreateInstanceConstraint is not needed for the neptune implementation, as constraints are
+// not a neptune construct
```

Comment:
Consider maybe returning an error instead of nil here. Returning `nil` allows this to be used and fail silently in the background whereas returning something like
```
errors.New("method not supported: CreateInstanceConstraint")
```
Would make this very obvious and give us clear visibility that we were using something that we're not supposed to.